### PR TITLE
GH23: Netstandard1.6 update

### PR DESCRIPTION
**Pull request by phenixdotnet**
Hi.

Please find a PR with the project migrated to new csproj format and with support of net45 and netstandard1.6 (branch name is incorrect I'm sorry)

Build parameters has been updated too.

A dependency to Newtonsoft.Json has been added to replace the default Json serializer implementation which is missing from the netstandard.
Cake dependency has been upgraded to latest stable (0.20.0)

Regards

### CODE REVIEW

File: setup.cake [hunk 1]
```diff
@@ -14,9 +14,10 @@ BuildParameters.PrintParameters(Context);
 
 ToolSettings.SetToolSettings(context: Context,
                             dupFinderExcludePattern: new string[] { 
-                                BuildParameters.RootDirectoryPath + "/src/Cake.Http.Tests/**/*.cs" },
+                                BuildParameters.RootDirectoryPath + "/src/Cake.Http.Tests/**/*.cs",
+								BuildParameters.RootDirectoryPath + "/**/Cake.Http.AssemblyInfo.cs"},
```
Comment (gep13) [inline setup.cake]: @phenixdotnet why was this addition required?

File: nuspec/nuget/Cake.Http.nuspec [hunk 2]
```diff
@@ -17,8 +17,14 @@
       <tags>cake script build httpclient web</tags>
    </metadata>
    <files>
-      <file src="Cake.Http.dll" target="lib/net45" />
-      <file src="Cake.Http.pdb" target="lib/net45" />
-      <file src="Cake.Http.xml" target="lib/net45" />
+      <file src="net45/Cake.Http.dll" target="lib/net45" />
```
Comment (gep13) [inline nuspec/nuget/Cake.Http.nuspec]: @phenixdotnet If you are going to continue using this nuspec file, then the setting for ShouldRunDotNetPack should be set to false in the setup.cake file, so that there isn't duplication of effort.

**Comment by gep13** [comment]
@phenixdotnet the build failure is likely due to requiring this:

https://github.com/cake-contrib/Cake.HockeyApp/blob/develop/.appveyor.yml#L4

in the appveyor.yml file.

File: setup.cake (hunk 1 above)
Comment (phenixdotnet) [inline setup.cake]: Because there is a tool which generate the Cake.Http.AssemblyInfo.cs for both net45 and netstandard1.6 and dupFinder tool doesn't like it :)

File: nuspec/nuget/Cake.Http.nuspec (hunk 2 above)
Comment (phenixdotnet) [inline nuspec/nuget/Cake.Http.nuspec]: According to [parameters.cake](https://github.com/cake-contrib/Cake.Recipe/blob/develop/Cake.Recipe/Content/parameters.cake#L272) the default value for ShouldRunDotNetPack is false, so I think there is no changes to be done.

**Comment by phenixdotnet** [comment]
Thanks the link ! 
Appveyor is now happy with this PR :)

**Comment by phenixdotnet** [comment]
Don't merge this PR, it break cake scripts when trying to load the lib due to an incompatibility between the referenced System.Net.Http from NuGet and the version used by Cake.
```
Could not load file or assembly 'System.Net.Http, Version=4.1.1.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```

**Comment by phenixdotnet** [comment]
It should be ok now.

**Comment by gep13** [comment]
@louisfischer having two labels will not work if you are using GitReleaseManager to generate your release notes.  It expects that there is only one label.

**Comment by gep13** [comment]
@louisfischer also, why is this a breaking change?

**Comment by louisfischer** [comment]
@phenixdotnet Thanks for completing this
@gep13 Because of the dependency on Newsoft.Json. This caused issues in the past if someone was using a different version than the one referenced in Cake.Http

File: src/Cake.Http/Cake.Http.csproj [hunk 3]
```diff
@@ -1,64 +1,20 @@
...
-    <Reference Include="Cake.Core, Version=0.19.3.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\Cake.Core.0.19.3\lib\net45\Cake.Core.dll</HintPath>
-    </Reference>
-    <Reference Include="System" />
-    <Reference Include="System.Core" />
-    <Reference Include="Microsoft.CSharp" />
-    <Reference Include="System.Data" />
-    <Reference Include="System.Net.Http" />
-    <Reference Include="System.Web.Extensions" />
+    <PackageReference Include="Cake.Core" Version="0.20.0" />
```
Comment (pascalberger) [inline src/Cake.Http/Cake.Http.csproj]: Is Cake 0.20 really required to use this plugin?

File: src/Cake.Http/Cake.Http.csproj [hunk 4]
```diff
@@ -1,64 +1,20 @@
...
+    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
```
Comment (pascalberger) [inline src/Cake.Http/Cake.Http.csproj]: Without any ILMerging or embedding of references, this will make it impossible to use this addin if any other version of Json.Net is already loaded.

Comment (phenixdotnet) [inline src/Cake.Http/Cake.Http.csproj]: @pascalberger Agree with you, do you have any guidance to embed or merge IL ?

Comment (pascalberger) [inline src/Cake.Http/Cake.Http.csproj]: @phenixdotnet See https://github.com/cake-contrib/Home/blob/master/Guidelines.md#assembly-references.

I'm using Fody in some other addins. I think @louisfischer already tried once and run into some issues, but don't know any details.

**Comment by phenixdotnet** [comment]
@gep13 @louisfischer There is another problem with this PR, Newtonsoft.Json.dll and System.Runtime.Serialization.Formatters.dll are required to use the lib (if internal json serialization is used) but the ```dotnet pack``` command doesn't add it and Cake doesn't restore the dependencies.

In my cake script, I mitigated this by manually copying the required libs after restore but it's clearly not suitable for "production" usage.

File: src/Cake.Http/Cake.Http.csproj (hunk 4 above)
Comment (phenixdotnet) [inline src/Cake.Http/Cake.Http.csproj]: @pascalberger thanks for the link, I will try it next week.

**Comment by gep13** [comment]
@phenixdotnet that is correct.  `dotnet pack` will not include it.  If you continue to want to use the external dependency, then I would suggest **not** using `dotnet pack` and instead use the nuspec file.

**Comment by louisfischer** [comment]
@pascalberger You are correct. I ran into an issue even when using Fody. That is why I was going to research how replace the Json Serializer. I am open for suggestions.

**Comment by louisfischer** [comment]
We may have to wait until .Net Standard 2.0 RTMs. I know Json.Net makes it easy to convert from object to Json but I wanted (as a goal for this project) to limit the dependencies to Cake (Cake.Core). Json.Net is a very popular Json Formatter. As such, it is very likely that someone using it in their cake.build script files. I am not sure taking the dependency is worth the trouble. 

I may just want to remove the SetJsonRequestBody<T> method to taking the dependency Json.Net. It is relatively painless to format the data in the cake.build script.

@pascalberger What do you think? I believe you were the one who originally requested the method.

**Comment by phenixdotnet** [comment]
@louisfischer 

> I may just want to remove the SetJsonRequestBody method to taking the dependency Json.Net. It is relatively painless to format the data in the cake.build script.

I thinks its a good proposition, we can remove this method, lets users do the serialization in cake scripts and re-introduce it when .NET Standard 2.0 is out if the json data contract serializer is include.
